btrack/dataio.py

```diff
@@ -588,18 +588,36 @@ def write_properties(self, data: dict):
 
         if 'properties' not in grp.keys():
             props_grp = grp.create_group('properties')
+        else:
+            props_grp = self._hdf[f'objects/{self.object_type}']['properties']
 
         n_objects = len(self.objects)
 
         for key, values in data.items():
+            # Manage the property data:
             if not values:
                 logger.warning("Property {key} is empty.")
                 continue
             values = np.array(values)
+            assert values.shape[0] == n_objects
+
+            # Check if the property is already in the props_grp:
+            if key in props_grp:
+                if overwrite is False:
+                    logger.info(f"Property '{key}' already written in the file")
+                    raise TypeError(
+                        f"Property '{key}' already in file -> switch on "
+                        "'overwrite' param to replace existing property "
+                    )
+                    continue
```

Comment:
I'm not sure this continue will do anything if you raise the `TypeError` above. Do you want it to continue trying to write out?

Reply:
You could also raise a `KeyError` here instead